### PR TITLE
docs: write docs for new workflow

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

File: docs/cli-commands.md [hunk 1]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+
+The defaults are intended for local testing: 
+
+```bash
+dsp-tools ingest-xmlupload xml_data_file.xml
+```
+
+will create the resources contained in `xml_data_file.xml` on `localhost` for local viewing.
+
+In order to create the same resources on the DSP server `https://app.dasch.swiss`,
```
Comment (Nora-Olivia-Ammann) [inline docs/cli-commands.md:326]: Same as above.

File: docs/cli-commands.md [hunk 2]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+
+| <center>Warning</center>                                          |
+| ----------------------------------------------------------------- |
+| These commands are experimental. They might change in the future. |
+
+This new workflow consists of 3 commands:
+
+- [`upload-files`](#upload-files): upload all files that are referenced in an XML file to a DSP server
+- [`ingest-files`](#ingest-files): kick off the ingest process, and retrieve the mapping CSV when it has finished
+- [`ingest-xmlupload`](#ingest-xmlupload): create the resources contained in the XML file, using the mapping CSV
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply docs/cli-commands.md:221]: The `xmlupload` does many things, that's why its description is so generic (it says "This command uploads data defined in an XML file to a DSP server"). But now, we split this up into smaller bits, so each bits needs a more detailed/precise description.

I don't think that "upload the XML" is an accurate description of what `ingest-xmlupload` makes. I would say that "create the resources contained in the XML file" is a precise, understandable, correct description of what happens.

File: docs/cli-commands.md [hunk 3]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+
+
+### `upload-files`
+
+This command uploads all files contained in an XML file to a server, without any processing/ingesting.
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply docs/cli-commands.md:226]: This just uploads one file after the other, and prints a message for every uploaded file. When all files are uploaded, the process terminates. Alternatively, we could also use a progress bar.

File: docs/cli-commands.md [hunk 4]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+
+The defaults are intended for local testing: 
+
+```bash
+dsp-tools upload-files xml_data_file.xml
+```
+
+will upload the data defined in `xml_data_file.xml` on `localhost` for local viewing.
+
+In order to upload the same data to the DSP server `https://app.dasch.swiss`,
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply docs/cli-commands.md:249]: I think that every paragraph of this page should be understandable on its own. We cannot expect that the users have read the other paragraphs, and know how to use the `-s`, `-u`, and `-p` flags. That's why it's repeated in several paragraphs.

The current wording is: "The defaults are intended for local testing (...) In order to upload the same data to the DSP server `https://app.dasch.swiss`, it is necessary to specify the following options". Do you think we should adapt this wording?

An alternative would be that we remove this explanation altogether, but my impression is that RDU members (especially unexperienced and/or new RDU members) need it.

File: docs/cli-commands.md [hunk 5]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+- `--suppress-update-prompt` (optional): don't prompt when using an outdated version of DSP-TOOLS 
+  (useful for contexts without interactive shell, e.g. when the Terminal output is piped into a file)
+
+The defaults are intended for local testing: 
+
+```bash
+dsp-tools upload-files xml_data_file.xml
+```
+
+will upload the data defined in `xml_data_file.xml` on `localhost` for local viewing.
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply docs/cli-commands.md:247]: better now?

File: docs/cli-commands.md [hunk 6]
```diff
@@ -208,6 +208,130 @@ Currently, only one interrupted upload can be resumed at a time per server.
...
+dsp-tools xmlupload -s https://api.dasch.swiss -u 'your@email.com' -p 'password' xml_data_file.xml
+```
+
+The expected XML format is [documented here](./file-formats/xml-data-file.md).
+
+
+### `ingest-files`
+
+This command kicks off the ingest process on the server, waits until it has completed, 
+and then saves the mapping CSV in the current working directory.
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply docs/cli-commands.md:262]: better now?

File: docs/cli-commands.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nora-Olivia-Ammann) [reply docs/cli-commands.md:221]: To me it sounds a bit too technical.

File: docs/cli-commands.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nora-Olivia-Ammann) [reply docs/cli-commands.md]: Progress bar sounds good!

File: docs/cli-commands.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Nora-Olivia-Ammann) [reply docs/cli-commands.md:250]: I just saw that this is the standard with all the other commands so for consistency's sake we can leave it.

File: docs/cli-commands.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/cli-commands.md:221]: It sounds pretty technical, which I think is ok as long as this is an experimental feature. And when we stabilize it, I envision us having a wrapper command that does these steps one after another, so users should not have to care about these steps. So I would say, we can optimize for approachability of the docs when we get there.

**Review by BalduinLandolt — APPROVED** [review]
only skimmed over this one, but looks good to me

**Review by Nora-Olivia-Ammann — APPROVED** [review]

**Pull request merged**